proposals/2117-marquez-over-time.md

```diff
@@ -0,0 +1,34 @@
+# Proposal: Marquez API to retrieve historical data
+
+## Problem
+
+Marquez data model in PostgresSQL allows extracting a snapshot of any lineage content from the past. This may be extremely
+useful to track how job, dataset etc. evolved over time. The feature can be implemented within each Marquez endpoint: like
+retrieving datasets or jobs. The purpose of this doc is to agree on a consistent API implementation approach. 
+
+## Solution
+
+Point in time can be described by a datetime, a dataset version or a job version. In case of dataset version or job version, an API should find out 
```

Comment:
IMO, versions are the only reliable parameter for a point in time lineage API. I think a good UI might give users the option to scan versions by some time condition (e.g., I want to diagnose the job run that executed yesterday at 12pm - tell me what job version was executed and what dataset version was read). Datetime as a parameter is ambiguous. If a datetime falls between two dataset versions, which one should be picked? Does the parameter have to match the createdAt field exactly? If so, why not just use the version?

Reply:
A runId is also a good input parameter for point-in-time lineage, as a run is always tied to a specific job version and dataset version (at least, there's only ever one version for a given output dataset in a run).

Reply:
>If a datetime falls between two dataset versions, which one should be picked? 

I think here the choice is for version that exists - meaning older one. However, I agree that having support from UI side for this is superior choice.

Reply:
ok, let's restrict the API to version parameters.